src/functional/fencil_processors/source_modules/cpp_gen.py

```diff
@@ -14,15 +14,35 @@
 
 
 import ctypes
+from dataclasses import dataclass
 from types import MappingProxyType
 from typing import Final, Sequence, Type
 
 import numpy
 
-from functional.fencil_processors import source_modules
+from eve.codegen import format_source
+from functional.fencil_processors.source_modules import source_modules
 
 
-LANGUAGE_ID: Final = "cpp"
+@dataclass(frozen=True)
+class CppLanguage:
+    """Implements source_modules.IncludeImplementationLanguageProtocol for C++ language settings."""
+
+    name: str
```

Comment:
If we identify languages by the class type, `name` here is redundant. If we still want a string name for messages to the user, it might be nicer to use pretty names like `"C++"` and `"Python"`.